### PR TITLE
feat: enable trustedTypes for html

### PR DESCRIPTION
**Pull request by Jack-Works**
- chore: reduce use of dangerouslySetInnerHTML
- feat: enable trusted types for html

## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/polyfills/types/dom.d.ts [hunk 1]
```diff
@@ -13,10 +12,54 @@ interface ImportMeta {
...
     adoptedStyleSheets?: readonly CSSStyleSheet[]
 }
 interface CSSStyleSheet {
     replace(text: string): Promise<void>
     replaceSync(text: string): void
 }
+// #endregion
+
+// #region Trusted Types
+interface TrustedHTML {
```
Comment (septs) [inline packages/polyfills/types/dom.d.ts:26]: need eslint rule? prefer-trusted-html

Comment (Jack-Works) [reply packages/polyfills/types/dom.d.ts:26]: No need, and you also cannot make a linter rule for it.

File: packages/mask/.webpack/manifest.ts [hunk 2]
```diff
@@ -3,18 +3,39 @@ import emitFile from '@nice-labs/emit-file-webpack-plugin'
 import { cloneDeep } from 'lodash-unified'
 import { NormalizedFlags } from './flags'
 
-type Manifest = typeof import('../src/manifest.json') & Record<string, any>
+type ManifestV2 = typeof import('../src/manifest.json') & { key?: string }
+type ManifestV3 = typeof import('../src/manifest-v3.json') & { key?: string }
+
 export function emitManifestFile(flags: NormalizedFlags) {
     return emitFile({
```
Comment (septs) [inline packages/mask/.webpack/manifest.ts:10]: suggest, use `emitJSONFile` replace it.

see https://github.com/NiceLabs/emit-file-webpack-plugin/blob/617c374a84b4f043ae15d087d2f3f999927b4a33/index.ts#L58-L65

Comment (Jack-Works) [reply packages/mask/.webpack/manifest.ts:10]: I don't know if we can do that. Some of our release processes may involve editing `../src/manifest.json`

**Comment by Jack-Works** [comment]
Merge conflict fixed.

**Pull request merged**